### PR TITLE
:book: Add pkg.go.dev badge and update package versions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
#### What kind of change does this PR introduce?
   Doc update

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

### CODE REVIEW

**Comment by azeemshaikh38** [comment]
I'm concerned about directing users here while we don't officially yet support Scorecard as a library. Apart from the LGTM.

**Comment by naveensrinivasan** [comment]
> I'm concerned about directing users here while we don't officially yet support Scorecard as a library. Apart from the LGTM.

I think we should start supporting it. There are projects using it. 

- Scorecard-actions (this is WIP)
- Witness 
- Allstar

We should document that we don't support Semver but `scorecard/v4` similar to Go GitHub Libary. 

Thoughts? @azeemshaikh38 @laurentsimon @justaugustus

File: README.md
Comment (justaugustus) [inline README.md:7]:
```suggestion
[![Go Reference](https://pkg.go.dev/badge/github.com/ossf/scorecard/v4.svg)](https://pkg.go.dev/github.com/ossf/scorecard/v4)
[![Go Report Card](https://goreportcard.com/badge/github.com/ossf/scorecard/v4)](https://goreportcard.com/report/github.com/ossf/scorecard/v4)
```

**Review by justaugustus — COMMENTED** [review]
Adding code suggestions on package version and pkg.go.dev URL.

**Review by justaugustus — APPROVED** [review]

**Comment by justaugustus** [comment]
> > I'm concerned about directing users here while we don't officially yet support Scorecard as a library. Apart from the LGTM.
> 
> I think we should start supporting it. There are projects using it.

This presentation is worth a look: https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#apidesign

Of note:

> If it’s exported, its part of your public API.

That said, we have a responsibility to [design APIs that are hard to misuse](https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_design_apis_that_are_hard_to_misuse).

>     * Scorecard-actions (this is WIP)

https://github.com/ossf/scorecard-action/issues/107 (cc: @rohankh532)

>     * Witness

https://github.com/ossf/scorecard/issues/1673 / https://github.com/ossf/scorecard/issues/1245 / https://github.com/ossf/scorecard/pull/1682 (cc: @colek42)

>     * Allstar

https://github.com/ossf/scorecard/pull/1645 / https://github.com/ossf/allstar/pull/114

https://github.com/ossf/scorecard/pull/1645 just merged, which makes some decent steps in the right direction and I'll open a separate tracking issue for this if one doesn't already exist.

> We should document that we don't support Semver but `scorecard/v4` similar to Go GitHub Libary.

Disagree here and I've expressed the opinion in a [few places](https://github.com/ossf/scorecard/issues/1621#issuecomment-1040930556), including the most recent (2022-02-24) [biweekly meeting](https://docs.google.com/document/d/1dB2U7_qZpNW96vtuoG7ShmgKXzIg6R5XT5Tc-0yz6kE/edit#bookmark=id.4o9aq4yyzl1s).

We've adopted SemVer-compliant versioning, which means regardless if we've enforce SemVer or not, consumers will have at least a baseline expectation that we are compliant. Whether that is true or not is a different story.

I think we should instead declare that the next major release version ([v5.0.0](https://github.com/ossf/scorecard/issues/1490)) is the first release in which we will be SemVer-compliant and direct our efforts into designing the module in a way that we can ensure that.

I think we'll all agree that scorecard is a high-value project in a space with lots of eyes on it today, so we have a responsibility to those who consume it to make it as easy and clear to do so as possible.

**Comment by justaugustus** [comment]
> I think we should instead declare that the next major release version ([v5.0.0](https://github.com/ossf/scorecard/issues/1490)) is the first release in which we will be SemVer-compliant and direct our efforts into designing the module in a way that we can ensure that.

I've created a tracking issue for this and added it to the v5 milestone: https://github.com/ossf/scorecard/issues/1683
Let's continue the discussion there!

**Comment by naveensrinivasan** [comment]
> > > I'm concerned about directing users here while we don't officially yet support Scorecard as a library. Apart from the LGTM.
> 
> > 
> 
> > I think we should start supporting it. There are projects using it.
> 
> 
> 
> This presentation is worth a look: https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#apidesign
> 
> 
> 
> Of note:
> 
> 
> 
> > If it’s exported, its part of your public API.
> 
> 
> 
> That said, we have a responsibility to [design APIs that are hard to misuse](https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_design_apis_that_are_hard_to_misuse).
> 
> 
> 
> >     * Scorecard-actions (this is WIP)
> 
> 
> 
> https://github.com/ossf/scorecard-action/issues/107 (cc: @rohankh532)
> 
> 
> 
> >     * Witness
> 
> 
> 
> https://github.com/ossf/scorecard/issues/1673 / https://github.com/ossf/scorecard/issues/1245 / https://github.com/ossf/scorecard/pull/1682 (cc: @colek42)
> 
> 
> 
> >     * Allstar
> 
> 
> 
> https://github.com/ossf/scorecard/pull/1645 / https://github.com/ossf/allstar/pull/114
> 
> 
> 
> https://github.com/ossf/scorecard/pull/1645 just merged, which makes some decent steps in the right direction and I'll open a separate tracking issue for this if one doesn't already exist.
> 
> 
> 
> > We should document that we don't support Semver but `scorecard/v4` similar to Go GitHub Libary.
> 
> 
> 
> Disagree here and I've expressed the opinion in a [few places](https://github.com/ossf/scorecard/issues/1621#issuecomment-1040930556), including the most recent (2022-02-24) [biweekly meeting](https://docs.google.com/document/d/1dB2U7_qZpNW96vtuoG7ShmgKXzIg6R5XT5Tc-0yz6kE/edit#bookmark=id.4o9aq4yyzl1s).
> 
> 
> 
> We've adopted SemVer-compliant versioning, which means regardless if we've enforce SemVer or not, consumers will have at least a baseline expectation that we are compliant. Whether that is true or not is a different story.
> 
> 
> 
> I think we should instead declare that the next major release version ([v5.0.0](https://github.com/ossf/scorecard/issues/1490)) is the first release in which we will be SemVer-compliant and direct our efforts into designing the module in a way that we can ensure that.
> 
> 
> 
> I think we'll all agree that scorecard is a high-value project in a space with lots of eyes on it today, so we have a responsibility to those who consume it to make it as easy and clear to do so as possible.

I agree. Thanks for the details. 

I will wait for @azeemshaikh38 before I merge.

**Comment by justaugustus** [comment]
> I will wait for @azeemshaikh38 before I merge.

I agree that the API is a longer discussion, but I don't think we need to block the badge updates on it.

**Comment by naveensrinivasan** [comment]
> > I will wait for @azeemshaikh38 before I merge.
> 
> I agree that the API is a longer discussion, but I don't think we need to block the badge updates on it.

Sounds good. 👍

**Pull request merged**
